File: porting-from-algo-c/src/lissajouscurve.lua
--
--	from src/lissaj.c
--
--	a part of main		to	lissajousCurve
--

local PI = math.pi

local m_sin = math.sin
local m_cos = math.cos

local function lissajousCurve(plotter, n, offset)
	function step(a, b)
		return n + offset + n * m_cos(a), n + offset + n * m_sin(b)
	end

	plotter:move(step(0, 0))
	for i=1,360 do
		plotter:draw(step(3 * (PI / 180) * i, 5 * (PI / 180) * i))
	end
end

return {
	lissajousCurve = lissajousCurve
}

File: porting-from-algo-c/src/lissajouscurve.py
#
#	from src/lissaj.c
#
#	a part of main		to	lissajousCurve
#

import math

def _stepX(n, offset, a):
    return n + offset + n * math.cos(a)

def _stepY(n, offset, a):
    return n + offset + n * math.sin(a)

def lissajousCurve(plotter, n, offset):
    plotter.move(_stepX(n, offset, 0), _stepY(n, offset, 0))
    for i in range(1, 361):
        plotter.draw(_stepX(n, offset, 3 * (math.pi / 180) * i),
                     _stepY(n, offset, 5 * (math.pi / 180) * i))

def _demo(path):
    import svgplot

    with open(path, "w") as fh:
        size, offset = 300, 10
        plotter = svgplot.svgPlot((size + offset) * 2, (size + offset) * 2)
        plotter.plotStart(fh)
        plotter.pathStart()
        lissajousCurve(plotter, size, offset)
        plotter.pathEnd()
        plotter.plotEnd()

if __name__ == "__main__":
    _demo("results/lissajouscurve-py.svg")
